src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -1371,13 +1025,9 @@ public static int IndexOf<T>(T[] array, T value, int startIndex, int count)
                 return (result >= 0 ? startIndex : 0) + result;
             }
 
-            return EqualityComparer<T>.Default.IndexOf(array, value, startIndex, count);
+            return IndexOfImpl(array, value, startIndex, count);
```

Comment:
These extra layers of generic calls can overhead that is pretty noticeable for methods that do very little work otherwise. For example, the following runs about 25% slower with this change:

```
struct S<T> : IEquatable<S<T>> where T:class
{
    T _o;

    public bool Equals(S<T> other) => _o == other._o;
}

S<object>[] a = new S<object>[10];
for (int i = 0; i < 1000000000; i++)
    Array.IndexOf(a, default(S<object>), 0, 2); 
```

Could you please omit the changes that add these extra layers of generics from this change, so that we can look at them separately?

Reply:
Done